test/test_helper.exs

```diff
@@ -11,3 +11,16 @@ case :gen_tcp.connect('localhost', 4222, [:binary]) do
               " #{:inet.format_error(reason)}\n" <>
               "You probably need to start gnatsd."
 end
+
+case :gen_tcp.connect('localhost', 4223, [:binary]) do
+  {:ok, socket} ->
+    :gen_tcp.close(socket)
+  {:error, reason} ->
+    Mix.raise "Cannot connect to gnatsd" <>
+              " (http://localhost:4223):" <>
```

Comment:
I like the helpful error message here, but it seems like a pain to force all developers to run two gnatsd servers while developing. Maybe this is a good time to tag our tests and default the integration tests to be skipped